lib/dfe/analytics/load_entity_batch.rb

```diff
@@ -4,27 +4,39 @@ class LoadEntityBatch < AnalyticsJob
       # https://cloud.google.com/bigquery/quotas#streaming_inserts
       # at a batch size of 500, this allows 20kb per record
       BQ_BATCH_MAX_BYTES = 10_000_000
+      MIN_BATCH_SIZE = 1
 
       def perform(model_class_arg, ids, entity_tag)
-        model_class = resolve_model_class(model_class_arg)
+        return if ids.empty?
 
-        events = create_events(model_class, ids, entity_tag)
+        model_class = resolve_model_class(model_class_arg)
 
-        payload_byte_size = events.sum(&:byte_size_in_transit)
+        # Process the entire batch if it's small enough or if splitting is no longer feasible
+        if ids.size <= MIN_BATCH_SIZE
+          process_batch(model_class, ids, entity_tag)
+        else
+          events = create_events(model_class, ids, entity_tag)
+          payload_byte_size = events.sum(&:byte_size_in_transit)
 
```

Comment:
I introduced MIN_BATCH_SIZE here because with the updated data structure, I was getting recursion between ln 16 and 19. Introducing this seemed to fix it